wallet/src/main/java/com/android/identity_credential/wallet/credentialoffer/CredentialOfferIssuance.kt

```diff
@@ -0,0 +1,51 @@
+package com.android.identity_credential.wallet.credentialoffer
+
+import com.android.identity.document.DocumentStore
+import com.android.identity.issuance.remote.WalletServerProvider
+import com.android.identity_credential.wallet.ProvisioningViewModel
+import com.android.identity_credential.wallet.SettingsModel
+import com.android.identity_credential.wallet.navigation.WalletDestination
+import org.json.JSONObject
+
+
+/**
+ * Initiate the process of setting a dynamic Credential Issuer defined in the credential offer
+ * payload that produces [credentialIssuerUri] Url of Issuer for obtaining credentials and
+ * [credentialIssuerConfigurationId] for the Credential Id (such as "pid-mso-mdoc" or "pid-sd-jwt").
+ */
+fun initiateCredentialOfferIssuance(
+    walletServerProvider: WalletServerProvider,
+    provisioningViewModel: ProvisioningViewModel,
+    settingsModel: SettingsModel,
+    documentStore: DocumentStore,
+    onNavigate: (String) -> Unit,
+    credentialIssuerUri: String,
+    credentialIssuerConfigurationId: String,
+) {
+    provisioningViewModel.start(
+        walletServerProvider = walletServerProvider,
+        documentStore = documentStore,
+        settingsModel = settingsModel,
+        issuerIdentifier = null,
+        credentialIssuerUri = credentialIssuerUri,
+        credentialIssuerConfigurationId = credentialIssuerConfigurationId
+    )
+    onNavigate(WalletDestination.ProvisionDocument.route)
+}
+
+/**
+ * Parse the Url Query component from the credential offer / haip Url and return a Pair containing
+ * the Credential Issuer Uri and Credential (Config) Id. This function produces the input parameters
+ * for initiating the Credential Offer Issuance flow.
+ */
+fun extractCredentialIssuerData(urlQueryComponent: String): Pair<String, String> {
+    val jsonPayload = JSONObject("{$urlQueryComponent}")
+    val credentialOffer = jsonPayload.getJSONObject("credential_offer")
+    // extract Credential Issuer Uri and Credential (Config) Id
+    val credentialIssuerUri = credentialOffer.getString("credential_issuer")
+    val credentialConfigurationIds =
+        credentialOffer.getJSONArray("credential_configuration_ids")
+    // TODO should there be a future implementation addressing all specified ids in credential_configuration_ids ?
```

Comment:
Great point and normally I would do what you're suggesting (provide the mechanism for supporting more than 1 ID inherently even if only 1 is being used at the moment so we don't make changes to unrelated features/functions in the future if we end up supporting more than 1 ID).

However David's (wise) words echo in my ears: less is more (for this project). Until we know we have to implement multiple IDs, it makes more sense (time, effort, complexity of code) to get the feature up and running with the best fastest approach. I think supplying a `List<String>` (instead of `String`) would mean unnecessary scaffolding and boxing/unboxing of that 1 ID of the list when we can just reference the String. 

Happy to implement your suggestion if you think it makes more sense to do so!

Reply:
Fair enough. This is fine for now. We can revisit later if needed.